### DISCUSSION TITLE
Lay z spa St. Moritz #54175

### DISCUSSION THREAD

**Discussion by sandy1981Nett** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#95] [Q&A]
First of all, thank you for the great project! Can I also use this with the pump model # 54175?

Lay-Z-Spa St.Moritz

I opened the pump and there are 6 cables.

**Comment by visualapproach** [comment]
Hi! You're welcome! I haven't heard of any 6-pin model that doesn't work so far. New models may require three extra resistors. Every model is not tested ofc, there is very many, but they seem to follow the same architecture. Worst case I guess, is we need to make a new button map but probably one of the existing three maps work.

**Reply by sandy1981Nett** [reply]
I soldered everything together, but after the start (hello + ip is shown in the LCD) only the temperatures are shown in the display via the web interface or the control panel cannot be set.

I soldered in the 10k resistor, was that wrong?

**Reply by visualapproach** [reply]
Did you do this?
```
//uncomment your model and comment out the rest
#define PRE2021			//the older one, no hydrojets
//#define MIAMI2021		//no hydrojets
//#define MALDIVES2021	//hydrojets
```

The resistor is fine

**Reply by sandy1981Nett** [reply]
no, i'm checking that tonight.

**Reply by sandy1981Nett** [reply]
I used the standard setting, that is for models 2021?

**Reply by sandy1981Nett** [reply]
hi,

the Setting was 

//uncomment your model and comment out the rest
#define PRE2021			//the older one, no hydrojets
//#define MIAMI2021		//no hydrojets
//#define MALDIVES2021	//hydrojets

now testing this 

//uncomment your model and comment out the rest
//#define PRE2021			//the older one, no hydrojets
//#define MIAMI2021		//no hydrojets
//#define MALDIVES2021	//hydrojets

**Reply by visualapproach** [reply]
Now you commented out everything, telling the computer that you don't have any model at all...

Lines beginning with `//` is a comment. Lines without `//` is compiled.

**Reply by sandy1981Nett** [reply]
//uncomment your model and comment out the rest
#define PRE2021 //the older one, no hydrojets
//#define MIAMI2021 //no hydrojets
//#define MALDIVES2021 //hydrojets

it shows the temperature on the server but i can still switch via the server and via the display on the pump.

**Reply by sandy1981Nett** [reply]
now ... show the display Hello and than the ip ... 

but then all buttons and the lcd display flash up and down ...

what is wrong?

**Reply by sandy1981Nett** [reply]
![Uploading 16267182053949221487973942544514.jpg…]()
![Uploading 16267182225861930288352281528593.jpg…]()

**Reply by sandy1981Nett** [reply]
I can Find the error

**Reply by sandy1981Nett** [reply]
i find the error it is the Board

one solder joint was not okay... 

thx i'am happy... now i config mqtt ... i hope it is fine ...

**Reply by visualapproach** [reply]
That is good

**Comment by sandy1981Nett** [comment]
Uploading VID_20210719_193735.mp4…

**Reply by chrstnmr** [reply]
Hey, which model is your pump? See sticker on the side of the pump

**Reply by sandy1981Nett** [reply]
Lay z spa St. Moritz #54175

**Reply by chrstnmr** [reply]
At the pump there is a specific model number for the pump since the pumps are similar between the different SPAs. For me it is interesting whether the pump is the same, since I have similar problems.

**Reply by sandy1981Nett** [reply]
![IMG_20210718_210943](https://user-images.githubusercontent.com/87618510/126302275-37e2e077-806d-4391-a8b4-c63df1ed2bcc.jpg)

look at picture, in my case it was a defective solder joint

**Reply by chrstnmr** [reply]
Ah, thanks. Sorry, I was assuming the St. Moritz was a 2021 model with the 2021 pump (since the name is similar to mine St. Lucia :D).
Good to hear, that it works for you :). 

I am having problems with mine, as no information from the pump is received or can be transferred (buttons or temperature). Not sure where the problem is. Maybe I'm going to resolder it from scratch to rule out any errors on solding (which I excluded in the beginning since I do not see any faulty soldings on the PCB).
